### PR TITLE
feat(escalating-v2): Emit escalating metrics

### PR DESCRIPTION
**Pull request by snigdhas**
Update the ESCALATING_ISSUES use case during post_process as we see new events come in.

### CODE REVIEW

File: src/sentry/tasks/post_process.py [hunk 1]
```diff
@@ -114,6 +116,22 @@ def _capture_event_stats(event: Event) -> None:
...
+    """
+    Update metrics for escalating issues when an event is processed.
+    """
+    metrics_backend = KafkaMetricsBackend()
+    metrics_backend.counter(
+        UseCaseID.ESCALATING_ISSUES,
+        org_id=event.project.organization_id,
+        project_id=event.project.id,
+        metric_name="event_ingested",
+        value=1,
```
Comment (snigdhas) [inline src/sentry/tasks/post_process.py:129]: This should get aggregated as part of the counter right?

Comment (ayirr7) [reply src/sentry/tasks/post_process.py:130]: yes, that's correct

**Review by NisanthanNanthakumar — APPROVED** [review]

File: src/sentry/tasks/post_process.py [hunk 2]
```diff
@@ -115,6 +117,22 @@ def _capture_event_stats(event: Event) -> None:
...
+        tags={"group": event.group_id},
+        unit=None,
```
[the remark this answers is not in the document]
Comment (snigdhas) [reply src/sentry/tasks/post_process.py:132]: it's the mri_unit in this template "<type>:<ns>/<name>@<unit>". I've mostly seen units of time (millisecond, second) here

File: src/sentry/tasks/post_process.py [hunk 3]
```diff
@@ -115,6 +117,22 @@ def _capture_event_stats(event: Event) -> None:
...
+    Update metrics for escalating issues when an event is processed.
+    """
+    metrics_backend = KafkaMetricsBackend()
+    metrics_backend.counter(
+        UseCaseID.ESCALATING_ISSUES,
+        org_id=event.project.organization_id,
+        project_id=event.project.id,
+        metric_name="event_ingested",
+        value=1,
+        tags={"group": event.group_id},
```
Comment (ayirr7) [inline src/sentry/tasks/post_process.py:131]: Is `event.group_id` a string?

Comment (ayirr7) [reply src/sentry/tasks/post_process.py:131]: Tag keys and values must both be strings to be compatible with the Kafka schema here https://github.com/getsentry/sentry-kafka-schemas/blob/main/schemas/ingest-metrics.v1.schema.json

File: src/sentry/tasks/post_process.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ayirr7) [reply src/sentry/tasks/post_process.py:132]: The unit for the metric you are emitting (whatever you are measuring). If there is no unit then `None` is fine

File: src/sentry/tasks/post_process.py [hunk 4]
```diff
@@ -16,6 +16,8 @@
 from sentry.issues.grouptype import GroupCategory
 from sentry.issues.issue_occurrence import IssueOccurrence
 from sentry.killswitches import killswitch_matches_context
+from sentry.sentry_metrics.kafka import KafkaMetricsBackend
```
Comment (onewland) [inline src/sentry/tasks/post_process.py:19]: @ayirr7 I think we should hide which backend gets used from the caller.

it should be something like

```python
from sentry.sentry_metrics import get_sentry_metrics_backend

# today this returns a KafkaMetricsBackend, but later it
# should be configurable from somewhere outside this file
metrics_backend = get_sentry_metrics_backend() 
```

This also makes it easier to only have one connection (pool) per process

Comment (lynnagara) [reply src/sentry/tasks/post_process.py:19]: Sentry has the service delegator / LazyServiceWrapper abstraction which is commonly used for this

**Pull request merged**